### PR TITLE
python3Packages: add Sanic HTTP Server (and dependencies)

### PR DESCRIPTION
**Pull request by thoughtpolice**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by thoughtpolice** [comment]
@GrahamcOfBorg build python3Packages.uvloop python3Packages.httptools python3Packages.sanic

File: pkgs/development/python-modules/sanic/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, buildPythonPackage, fetchPypi, pythonOlder
+, aiofiles, websockets, multidict, ujson, uvloop, httptools
+}:
+
+let
+  # Ideally, websockets 7.0 will be supported in the future, but
+  # for now, sanic requires 6.x
+  ws = websockets.overrideAttrs (_: {
```
Comment (FRidh) [inline pkgs/development/python-modules/sanic/default.nix]: We can't have a different version of a package in `python-packages.nix` as that will result in collisions.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/sanic/default.nix (hunk 1 above)
Comment (thoughtpolice) [reply pkgs/development/python-modules/sanic/default.nix]: Am I missing something here, or are you suggesting that `python-packages.nix` infrastructure cannot feature any version overrides in any expressions? That seems... extremely not-Nix like. Or are you worried that `ws` will end up pulled in by possible downstream users of `sanic` (due to `propagatedBuildInputs`)?

Comment (FRidh) [reply pkgs/development/python-modules/sanic/default.nix]: Python cannot handle having multiple versions of a module in a single closure. If someone wants to build a Python env with this package, and another package that uses the other version of `ws`, they will get a collision. Now, one could say that is a risk/problem for the user whenever they compose such environment, but that would severely reduce the quality of the package set.

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by thoughtpolice** [comment]
Unfortunately not. Sanic has moved to supporting websockets 0.7.x, but the same problem has cropped up yet again: websockets is now on 0.8.x, making these packages mutually exclusive in the same way.

I'm closing this since I have little hope this will be rectified anytime soon.

**Pull request closed without merging**